android-components/components/lib/state/src/main/java/mozilla/components/lib/state/ext/ComposeExtensions.kt

```diff
@@ -43,6 +44,35 @@ fun <S : State, A : Action, R> Store<S, A>.observeAsComposableState(map: (S) ->
     return state
 }
 
+/**
+ * Starts observing this [Store] and represents the mapped state (using [map]) via [ComposeState].
+ *
+ * Every time the mapped [Store] state changes, the returned [ComposeState] will be updated causing
+ * recomposition of every [ComposeState.value] usage.
+ *
+ * The [Store] observer will automatically be removed when this composable disposes or the current
+ * [LifecycleOwner] moves to the [Lifecycle.State.DESTROYED] state.
+ *
+ * @param initialValue Initial value emitted.
+ * @param map The applied function to produced the mapped value [R] from [S].
+ * @return A non nullable [ComposeState], making the api more reasonable for callers where the
+ * state is non null.
+ */
+@Composable
+fun <S : State, A : Action, R> Store<S, A>.observeAsComposableState(
```

Comment:
Hmm yeah i see your point and in most other projects `observeAsState` would read better as there would be only compose state. But since we also have `lib-state` it can be confusing for the reader. `ComposableState` suffix is just a bit more explicit.

But, to counter my own point above, since we are in the composable scope, it's clear, that here it means `androidx.compose.runtime.State`. I agree, we can shorten the name. I'll leave this open for a couple of days for more comments and hopefully we can take a call by Monday.